zoomable-image/sub-sampling-image/src/commonMain/kotlin/me/saket/telephoto/subsamplingimage/internal/tileGridGenerator.kt

```diff
@@ -65,6 +65,28 @@ internal fun ImageRegionTileGrid.Companion.generate(
   )
 }
 
+internal fun ImageRegionTileGrid.Companion.defaultMinSize(
+  viewportSize: IntSize,
+  unscaledImageSize: IntSize,
+): IntSize {
+  val minWidth: Int
+
+  var candidateWidth = unscaledImageSize.width
+  while (candidateWidth > viewportSize.width / 2) {
+    candidateWidth /= 2
+  }
+  minWidth = candidateWidth
+
+  val minHeight: Int
+  var candidateHeight = unscaledImageSize.height
+  while (candidateHeight > viewportSize.height / 2) {
+    candidateHeight /= 2
+  }
+  minHeight = candidateHeight
+
+  return IntSize(minWidth, minHeight)
+}
```

Comment:
Can we add some test cases to [ImageRegionTileGridGeneratorTest](https://github.com/saket/telephoto/blob/99b24b99e93cd753a6245c5535f5c475cafffe91/zoomable-image/sub-sampling-image/src/commonTest/kotlin/me/saket/telephoto/subsamplingimage/internal/ImageRegionTileGridGeneratorTest.kt#L22) to cover this?